### PR TITLE
SwiftMailer transport can register optional plugins

### PR DESCRIPTION
**Pull request by craig410**

### CODE REVIEW

File: src/DependencyFactory/SwiftMailerFactory.php [hunk 1]
```diff
@@ -47,8 +47,17 @@ public static function definitions()
...
-    {
+    /**
+     * @param array|null                  $relay
+     * @param \Swift_Events_EventListener ...$plugins
+     *
+     * @return \Swift_SmtpTransport
+     */
+    public static function buildSmtpTransport(
+        array $relay = NULL,
+        \Swift_Events_EventListener ...$plugins
```
Comment (acoulton) [inline src/DependencyFactory/SwiftMailerFactory.php:58]: I'm not certain if this signature works if there are no plugins? Off the top of my head can't remember how it behaves following an optional argument. Could you check?

**Review by acoulton — CHANGES_REQUESTED** [review]

**Comment by craig410** [comment]
Empty array apparently. So the foreach does nothing.

File: src/DependencyFactory/SwiftMailerFactory.php (hunk 1 above)
Comment (craig410) [reply src/DependencyFactory/SwiftMailerFactory.php:58]:
```php
php > function test(...$var){var_dump($var);}
php > test();
array(0) {
}
```

Comment (acoulton) [reply src/DependencyFactory/SwiftMailerFactory.php:58]: Awesome, LGTM then just needs adding to the changelog

**Review by acoulton — APPROVED** [review]

**Pull request merged**